sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -532,6 +532,20 @@ object GpuOverrides {
         // There are so many of these that we don't need to print them out.
         override def print(append: StringBuilder, depth: Int, all: Boolean): Unit = {}
       }),
+    // TODO: maybe we should apply the conversion between decimal32 and decimal64 to promote precision?
+    expr[PromotePrecision](
```

Comment:
This is the same as `KnownFloatingPointNormalized` or `KnownNotNull`.  In those cases we don't throw it away, we replace it with s GPU version that is essentially a noop.  It has little performance impact and lets us keep the same structure that spark has.

Reply:
Yes! And this PR has been refactored.